server/sonar-process-monitor/src/main/java/org/sonar/process/monitor/Monitor.java

```diff
@@ -97,12 +108,34 @@ public void start(List<JavaCommand> commands) {
   private void startProcesses() {
     // do no start any child process if not in state INIT or RESTARTING (a stop could be in progress too)
     if (lifecycle.tryToMoveTo(State.STARTING)) {
+      resetFileSystem();
+
       startAndMonitorProcesses();
       stopIfAnyProcessDidNotStart();
     }
   }
 
+  private void resetFileSystem() {
+    // since JavaLauncher depends on temp directory, which is reset below, we need to close it first
+    closeJavaLauncher();
+    try {
+      fileSystem.reset();
+    } catch (IOException e) {
+      // failed to reset FileSystem
+      throw new RuntimeException("Failed to reset file system", e);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS00112)